modules/registry/registry.go

```diff
@@ -205,6 +204,52 @@ func (c *RegistryContainer) PushImage(ctx context.Context, ref string) error {
 	return c.ImageExists(ctx, ref)
 }
 
+// PullImage pulls an image from an external Registry.
+// It will use the internal registry to store the image.
+// This function is helpful when you want to push an image to your Registry
+// instance made by testcontainer.
+func (c *RegistryContainer) PullImage(ctx context.Context, ref string) error {
+	dockerCli, err := testcontainers.NewDockerClientWithOpts(ctx)
+	if err != nil {
+		return fmt.Errorf("create docker client: %w", err)
+	}
+	defer dockerCli.Close()
+
+	pullOpts := image.PullOptions{
+		All: true,
+	}
+
+	output, err := dockerCli.ImagePull(ctx, ref, pullOpts)
+	if err != nil {
+		return fmt.Errorf("failed to push image %s: %w", ref, err)
+	}
+
+	_, err = io.ReadAll(output)
+	if err != nil {
+		return fmt.Errorf("failed to read output: %w", err)
```

Comment:
suggestion: ensure output is closed, copy to discard which is more efficient, and clean up error:
```suggestion
    defer output.Close() //nolint: errcheck
    
	_, err = io.Copy(io.Discard, output)
	if err != nil {
		return fmt.Errorf("read image pull output: %w", err)
```